pkg/cmd/roachtest/tests/failover.go

```diff
@@ -32,6 +32,13 @@ import (
 )
 
 func registerFailover(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:    "failover/partial-disconnect",
```

Comment:
We should have tests for multiple kinds of partial partitions. I enumerated the ones I think are important in [the doc](https://docs.google.com/document/d/134eVs69fVBhP-01bxfrdjPxz--drNjOQ8AI-7Z3UD6U/edit). Let's call this e.g. `failover/partial/lease-liveness` so that we can add additional kinds later.

Should we test both blackhole and TCP reset failure cases? I think blackhole is probably sufficient since that's what we're most likely to see in the wild, and the TCP reset case is strictly easier to handle, but we could consider adding it for completeness.